src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -668,24 +662,28 @@ export class SavedObjectsRepository {
       const { type, id, opensearchRequestIndex } = expectedResult.value;
       const doc = bulkGetResponse?.body.docs[opensearchRequestIndex];
       if (doc?.found) {
-        const transformedObject = this._serializer.rawToSavedObject(doc as SavedObjectsRawDoc);
-        const filteredWorkspaces = SavedObjectsUtils.filterWorkspacesAccordingToBaseWorkspaces(
-          options.workspaces,
-          transformedObject.workspaces
-        );
-        if (!filteredWorkspaces.length) {
-          errors.push({
-            id,
-            type,
-            error: {
-              ...errorContent(SavedObjectsErrorHelpers.createConflictError(type, id)),
-              // @ts-expect-error MultiGetHit._source is optional
-              ...(!this.rawDocExistsInNamespace(doc!, namespace) && {
-                metadata: { isNotOverwritable: true },
-              }),
-            },
-          });
+        let workspaceConflict = false;
+        if (options.workspaces) {
+          const transformedObject = this._serializer.rawToSavedObject(doc as SavedObjectsRawDoc);
+          const filteredWorkspaces = SavedObjectsUtils.filterWorkspacesAccordingToBaseWorkspaces(
+            options.workspaces,
+            transformedObject.workspaces
+          );
+          if (!filteredWorkspaces.length) {
+            workspaceConflict = true;
+          }
         }
+        errors.push({
```

Comment:
Is this means we will always return errors when document found?

Reply:
Yes. for create / bulkCreate, we will always return an error when workspace checked conflict.